setup/sql/5_4_hotfixes.php

```diff
@@ -1248,5 +1248,38 @@
     $ilDB->execute($updateStatement, [$row['certificate_content'], $row['id']]);
 }
 ?>
+<#83>
+<?php
+$checked_files = [];
+$defaulf_bg_img_path = '/certificates/default/background.jpg';
+$checked_files[$defaulf_bg_img_path] = file_exists(CLIENT_DATA_DIR . $defaulf_bg_img_path);
+
+$res = $ilDB->query('SELECT id, certificate_content_bu FROM il_cert_user_cert WHERE background_image_path = "[BACKGROUND_IMAGE]"');
+$updateStatement = $ilDB->prepareManip("UPDATE il_cert_user_cert SET background_image_path = ? WHERE id = ?", ['text', 'integer']);
+while ($row = $ilDB->fetchAssoc($res)) {
+
+    $found = preg_match('/src="url\((.*\/certificates\/[a-zA-Z0-9\/]+\.[a-zA-Z]{3,4})\)/', $row['certificate_content_bu'], $matches);
+	$background_image_path = '';
+
+	if ($found !== false) {
+        $tmp_img_path = $matches[1];
+		$tmp_img_path = preg_replace('/.*(\/(assessment|course|exercise))/', '$1', $tmp_img_path);
+		$tmp_img_path = preg_replace('/.*(\/certificates\/(default|scorm))/', '$1', $tmp_img_path);
+
+		if (strlen($tmp_img_path) > 0) {
+			if (!array_key_exists($tmp_img_path, $checked_files)) {
+				$checked_files[$tmp_img_path] = file_exists(CLIENT_DATA_DIR . $tmp_img_path);
+            }
+
+            if ($checked_files[$tmp_img_path] === true) {
+                $background_image_path = $tmp_img_path;
+            }
+            // @todo if file does not exists, should the field be empty or set to default path?
```

Comment:
Yes, this is a good quesion. Maybe this has to be discussed by the JF.

Reply:
After I spoke with our project managers, we would prefer to set the path empty. Some of our customers use certificates with different company headers, because the learner are from sub-companies and shall get certificates "branded" by their company. In such cases it could be a bigger problem to get the wrong background instead of getting none.
Also I thought of logging this special cases (maybe on a new db table or column), to have some overview of certificates that must be fixed manually by re-uploading the image and correcting the database entry.